### ISSUE TITLE
Heroku Deploy notifications not showing up in Github Slack once upgraded to the new version

### ISSUE THREAD

**Issue by tonydehnke** [integrations/slack#1167]
In the Github Legacy,  we would get confirmation/fail notifications posted to the GitHub channel from Github when a deploy completed or failed (we use Heroku).   The new version does not do that anymore.

Anyone got it working? 

Sample of old output:
![image](https://user-images.githubusercontent.com/36720180/114990404-a8eed300-9ec2-11eb-98a6-415e86476f92.png)

**Comment by flaviobtc** [comment]
Same here!

**Comment by sjagoe** [comment]
We also use deployment notifications in some of our most important deployment workflows.

The feature disappearing so suddenly, apparently without warning, is now leading to people creating multiple concurrent deployments due to lack of feedback on the ongoing deploy process.

**Comment by sjagoe** [comment]
The release/migration notes for the updated github slack app have no mention of removed deployment notifications: https://github.com/integrations/slack#3-what-changes-after-the-upgrade

Specifically, it says:
> No change in the way our GitHub app for Slack works

It does say
> Removed deploy command support

But nowhere does it mention that already-subscribed notifications will be silently removed.

**Comment by ashokirla** [comment]
@tonydehnke @sjagoe @flaviobtc : Thanks for being our customer and we apologize for the inconvenience. We have been hearing number issues with the current deployment support and the usage is also very low. This is only a momentary change and we want to build a complete experience. If you have any concerns, we have happy to discuss over a call.

**Comment by mjuarez** [comment]
Not using Heroku, but my company relies on the deploy notifications. I'm a little disappointed this wasn't communicated clearly in advance so that we could prepare an alternative solution.

**Comment by quidquid** [comment]
Yes, the deploy command + notifications were part of our workflow too and they are missed. The "No change in the way the integration works" is really rather misleading when a moment later it says "btw no more deploys. ☺️"  What about at least changing the upgrade note / recommendation to warn people about this?

**Comment by tonydehnke** [comment]
@ashokirla - It seems from posts here and in other threads that the #1 thing people used was the notifications about deploys.. not so much the deploy commands from Slack.  Honestly, I'd feel way more comfortable executing those in Github or from a real command line when it comes to that.   
But getting updates about Deploys is 1000% what I use and like Slack for.   

So my suggestion would be to focus on getting the deploy notifications working again ASAP, and then later if demand warrants - adding the deploy command abilities. 

Happy to hop on a call to discuss things if that would help.  But for me - if the status notification I posted in my first post worked again, that would likely solve 99% of concerns for me.

**Comment by sjagoe** [comment]
I never even considered notifications about deployments to be directly related to the slack `/deploy` command at all. We use deployments extensively ouside of slack integration, and the only thing we had through slack was notification on them.

I think it's hugely disengenoues to remove them as part of the `/deploy` command feature having issues.

**Comment by emresandikci** [comment]
it's so bad to hear that HEROKU CI/CD deploy notification removed, there are more than 10 repo was using that feature in our company, remove the feature is really a major change.

**Comment by emresandikci** [comment]
The new upgrade removed CI/CD deploy notifications, so many projects affected my current company's and my side projects, it made a major problem to remove this feature without letting know.  It's like delete production database 🗡️

**Comment by swanson** [comment]
I ran into this issue as well today. There are a number of potential workarounds but some required using Heroku Pipelines, adding a custom buildpack, or using an external add-on. My preferred option (aside from the old GitHub integration) would be to be the Heroku "Deploy Webhooks", however, the payload from Heroku does not match what Slack needs to use the built-in "Incoming Webhooks" Slack integration.

I ended up writing a small Ruby script that I added to the "Release Phase" on Heroku. This will run as part of the deploy process and send a webhook request to Slack.

Here is the script (`bin/notify-slack`):

```ruby
#!/usr/bin/env ruby
require 'net/http'
require 'net/https'
require 'uri'
require 'json'

sha = ENV['HEROKU_SLUG_COMMIT']
app = ENV['HEROKU_APP_NAME']
webhook_url = ENV['SLACK_WEBHOOK_URL']
github_url = "https://github.com/YOUR_ORG/YOUR_REPO/commit/#{sha}"

message = "Successfully deployed <#{github_url}|`#{sha.slice(0, 7)}`> to *#{app}*"

puts "-----> Notifying Slack"
puts message
puts "------"

uri = URI.parse(webhook_url)
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true

req = Net::HTTP::Post.new(uri.request_uri)
req.set_form_data(payload: {
    fallback: message,
    color: "good",
    fields: [ { title: "Deploy", value: message } ]
  }.to_json
)
http.request(req)
```

You'll need to create an "Incoming Webhook" in your Slack workspace and add the Webhook URL to the `SLACK_WEBHOOK_URL` environment variable in your Heroku app.

![image](https://user-images.githubusercontent.com/56947/115479592-b683c000-a216-11eb-9565-71d1983183e8.png)


Lastly, add (or update) your Release phase in the `Procfile`. In my case this was:

```yaml
# Procfile
web: ./bin/server
release: bundle exec rails db:migrate && bin/notify-slack
worker: bundle exec sidekiq -C config/sidekiq.yml
```

You can customize the "Incoming Webhook" to be named "Heroku" and add an appropriate icon. You can also modify the script as needed if you want it to display differently.

After hooking everything up, we've got Slack notifications restored for Heroku deploys. Hope this helps as a stop-gap until the old functionality can be restored!

![image](https://user-images.githubusercontent.com/56947/115479473-76bcd880-a216-11eb-9d16-c624d0779e60.png)

**Comment by keyuls** [comment]
This is the most important issue. We spent 2 days waiting for deployment updates just because of this.

**Comment by tw0517tw** [comment]
We used to see "deployed to gh-pages" notification but now its gone. We really hope it will coma back

**Comment by ashokirla** [comment]
Thanks everyone for the feedback. We will look into this ask on priority.

**Comment by jakedouglas** [comment]
We were also relying on this for visibility into deployments and invested a bunch of work into hooking up CodePipeline events to the Deployments API just for the Slack notifications. Would greatly appreciate this being addressed.

**Comment by sinsoku** [comment]
You can revert to the legacy integration by re-installing from the URL below.
`https://<workspace>.slack.com/apps/A8GBNUWU8-github-legacy`

**Comment by RoohbirSingh** [comment]
@sinsoku thanks for this. This worked for me ^

**Comment by ashokirla** [comment]
Hi, 
Thanks everyone for the feedback. We will add deployment notifications in the new version of Slack app. We are working on this ask on priority and it should be available very soon.

**Comment by tonydehnke** [comment]
Hi @ashokirla any update on this?

**Comment by dylandechant** [comment]
hey @ashokirla any update on this?

**Comment by sasharevzin** [comment]
It feels like @ashokirla is no longer in charge 🤔

**Comment by swanson** [comment]
My old deploy notifications started showing up again this afternoon!

**Comment by tonydehnke** [comment]
Just notice I got them today too!!! 

![image](https://user-images.githubusercontent.com/36720180/120912075-712c2c80-c6b6-11eb-8a40-f88c0c06c4f3.jpeg)

**Comment by ashokirla** [comment]
Hi Everyone,

Yes, as we have promised, we have brought back the deployment notifications. Additionally, we have also handled deployment waiting scenarios where you have any approval or wait timer set on your environment which was not supported in old app. It has been deployed few days back. Since it is a feedback, we couldn't do another changelog.

Just like the old flow, deployment notifications is a default enabled feature i.e. you will receive deployment notifications once you subscribe to a repo. So, incase you don't want deployment notifications, you might have to unsubscribe the feature explicitly. This is a one time thing and we couldn't avoid this. We apologize for any inconvenience caused.
Incase if you are currently on old app and now you are planning to upgrade, the old configuration will be carried over i.e. you will not get deployment notifications if you unsubscribed for them in the old app. But if you are already on new app, you need to unsubscribe again.

Let us know if you have any queries.

Thanks

**Comment by motymichaely** [comment]
Hey there!  I've been facing the same issues and although the deployment notifications are back, I've created a new Heroku add-on that brings Heroku app change notifications to Slack (and also Webhooks and Microsoft Teams).  Would love to get your feedback:

https://activitytogo.com

Thanks!
